api/api.go

```diff
@@ -544,34 +550,50 @@ func (api *Server) getActions(start uint64, count uint64) (*iotexapi.GetActionsR
 		return nil, status.Error(codes.InvalidArgument, "range exceeds the limit")
 	}
 
+	totalActions, err := api.bc.GetTotalActions()
+	if err != nil {
+		return nil, status.Error(codes.Internal, err.Error())
+	}
+	if start >= totalActions {
+		return nil, status.Error(codes.InvalidArgument, "start exceeds the limit")
+	}
+
 	var res []*iotexapi.ActionInfo
-	var actionCount uint64
-	tipHeight := api.bc.TipHeight()
-	for height := 1; height <= int(tipHeight); height++ {
-		blk, err := api.bc.GetBlockByHeight(uint64(height))
+	var hit bool
+	for height := uint64(1); height <= api.bc.TipHeight(); height++ {
+		blk, err := api.bc.GetBlockByHeight(height)
 		if err != nil {
 			return nil, status.Error(codes.NotFound, err.Error())
 		}
 		selps := blk.Actions
+		if !hit && start >= uint64(len(selps)) {
+			start -= uint64(len(selps))
+			continue
+		}
```

Comment:
Shouldn't we set hit to true before the for loop?

Reply:
since start > len(selps), hit remains false before the for loop below